### PR TITLE
Convert from in-house animation to Revery's animation framework

### PR DESCRIPTION
**Pull request by glennsl**
This also refactors the Hover functionality, removing the models, fixing the animation being reset on every cursor move, and making it more or less ready for multiple cursors.

### CODE REVIEW

**Comment by bryphe** [comment]
I'm seeing an empty 'hover' square show up when testing this:

__Repro:__
- Open `Notifications.re`
- Move cursor (`jjjw`)

image

File: src/Store/LanguageFeatureConnector.re [hunk 1]
```diff
@@ -103,56 +102,32 @@ let start = () => {
   let updater = (state: Model.State.t, action: Actions.t) => {
     let default = (state, Isolinear.Effect.none);
     switch (action) {
-    | Actions.Tick({deltaTime, _}) =>
```
Comment (bryphe) [inline src/Store/LanguageFeatureConnector.re]: Nice to see this go away!

File: src/UI/HoverView.re [hunk 2]
```diff
@@ -3,97 +3,154 @@
...
+  let maybeEditor =
+    state |> Selectors.getActiveEditorGroup |> Selectors.getActiveEditor;
+
+  switch (maybeEditor) {
+  | Some(editor) when hoverEnabled && state.mode != Vim.Types.Insert =>
+    switch (Buffers.getBuffer(editor.bufferId, state.buffers)) {
+    | Some(buffer) =>
+      <View style=Styles.container>
+        {editor.cursors
+         |> List.map(cursor =>
```
Comment (bryphe) [inline src/UI/HoverView.re]: Interesting, once we have multi cursors we'll have hover support for each individual one!

File: src/UI/HoverView.re [hunk 3]
```diff
@@ -3,97 +3,154 @@
...
+    Configuration.getValue(c => c.editorHoverEnabled, state.configuration);
+
+  let State.{theme, editorFont, diagnostics, _} = state;
+  let maybeEditor =
+    state |> Selectors.getActiveEditorGroup |> Selectors.getActiveEditor;
+
+  switch (maybeEditor) {
+  | Some(editor) when hoverEnabled && state.mode != Vim.Types.Insert =>
+    switch (Buffers.getBuffer(editor.bufferId, state.buffers)) {
+    | Some(buffer) =>
```
Comment (bryphe) [inline src/UI/HoverView.re]: Related to the issue / screenshot I logged - seems like we're missing the case where there is no 'hover' - we're always showing at least a blank container w/ this change.

File: src/UI/QuickmenuView.re [hunk 4]
```diff
@@ -60,31 +60,36 @@ let onInputClicked = cursorPosition =>
...
+
+    <View style={Styles.progressBarIndicator(~width, ~offset, ~theme)} />;
+  };
+
+  <View style=Styles.progressBarTrack> <indicator /> </View>;
+};
+
+let%component busyBar = (~theme, ()) => {
+  let%hook (time, _, _) =
+    Animation.(
```
Comment (bryphe) [inline src/UI/QuickmenuView.re]: Cool!

File: src/UI/HoverView.re (hunk 3 above)
Comment (glennsl) [reply src/UI/HoverView.re]: Oh man, how did I miss that! Fixed in https://github.com/onivim/oni2/pull/1201/commits/d8b825b7f1290e0e7bb923694bb7c9d606dc937e

File: src/UI/HoverView.re (hunk 2 above)
Comment (glennsl) [reply src/UI/HoverView.re]: There's still the problem that it gets passed the x and y pixel coordinates of the cursor from EditorSurface. So right now the hover for every cursor would be rendered on top of the primary cursor. We'll need to pass in whatever is needed to calculate the pixel coordinates in `hoverItem`, and from a quick glance it seemed to need quite a few things.

**Pull request merged**